### ISSUE TITLE
LayerNorm behavior uncommon? (or wrong?)

### ISSUE THREAD

**Issue by albertz** [tensorpack/tensorpack#1482]
I noticed that in your LayerNorm implementation, you normalize over all axes except the first (batch-dim):
https://github.com/tensorpack/tensorpack/blob/ef0ecb8037e8dcdc83911600d7830da2553c273d/tensorpack/models/layer_norm.py#L42
```
mean, var = tf.nn.moments(x, list(range(1, len(shape))), keep_dims=True) 
```

This behavior is inconsistent to the normal standard behavior of layer-normalization, which just normalizes over the last (feature) axis. E.g. in TensorFlow ([here](https://github.com/tensorflow/tensorflow/blob/v2.3.0/tensorflow/python/keras/layers/normalization.py#L1203)), the code is:
```
mean, variance = nn.moments(inputs, self.axis, keep_dims=True)
```
where `axis` is -1 (or [-1] or [ndim - 1]) by default.

I think this default behavior as in TF is what you would usually expect, right? You also have the same e.g. in layer-normalization as used for the Transformer.

**Comment by ppwwyyxx** [comment]
> I think this default behavior as in TF is what you would usually expect, right?

No. Ours is the normal standard behavior for 4D input. Note that tensorFlow also uses it as the example: https://github.com/tensorflow/tensorflow/blob/b36436b087bd8e8701ef51718179037cccdfc26e/tensorflow/python/keras/layers/normalization.py#L1022-L1027
